### PR TITLE
 Feat: Pod Based Inclusions/Exclusions

### PR DESCRIPTION
**Pull request by stsantilena**
This PR adds the ability to include and exclude logs based on k8s pod name, namespace, labels and annotations. It adds two additional configuration variables:

`LOGDNA_K8S_METADATA_LINE_EXCLUSION`
`LOGDNA_K8S_METADATA_LINE_INCLUSION`

They syntax for including/excluding from various resources is as follows:
`namespace:<value>`
`name:<podname>`
`label.<key>:<value>`
`annotation.<key>:<value>`

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -320,6 +322,45 @@ Notes:
...
+* include *only* log lines associated with a specific resource with `LOGDNA_K8S_METADATA_INCLUSION`
+* exclude log lines associated with specific resource with `LOGDNA_K8S_METADATA_EXCLUSION`
+
+Currently, we support filtering on four different fields: namespace, pod, label and annotation. The following shows the nomenclature for each of the supported fields:
+
+```
+  namespace:<value>
+  pod:<value>
+  label.<key>:<value>
+  annotation.<key>:<value>"
```
Comment (jiangao1) [inline docs/README.md:340]: the extra " at the end of this line can be removed

Comment (stsantilena) [reply docs/README.md]: Fixed

File: docs/README.md [hunk 2]
```diff
@@ -320,6 +322,45 @@ Notes:
...
+You can configure the Kubernetes metadata filtering via the following environment variables:
+
+* include *only* log lines associated with a specific resource with `LOGDNA_K8S_METADATA_INCLUSION`
+* exclude log lines associated with specific resource with `LOGDNA_K8S_METADATA_EXCLUSION`
+
+Currently, we support filtering on four different fields: namespace, pod, label and annotation. The following shows the nomenclature for each of the supported fields:
+
+```
+  namespace:<value>
+  pod:<value>
```
Comment (stsantilena) [inline docs/README.md:338]: @c-nixon let me know your thought on using `pod` vs `name` based on our discussion around the metadata pod object.

File: docs/README.md [hunk 3]
```diff
@@ -191,6 +191,8 @@ options are available:
 |`LOGDNA_INCLUSION_REGEX_RULES`<br>**Deprecated**: `LOGDNA_INCLUDE_REGEX`|Comma separated list of regex patterns to include files from monitoring||
 |`LOGDNA_LINE_EXCLUSION_REGEX`|Comma separated list of regex patterns to exclude log lines. When set, the Agent will NOT send log lines that match any of these patterns.||
 |`LOGDNA_LINE_INCLUSION_REGEX`|Comma separated list of regex patterns to include log lines. When set, the Agent will send ONLY log lines that match any of these patterns.||
+|`LOGDNA_K8S_METADATA_EXCLUSION`|Comma separated list of Kubernetes selectors to exclude log lines associated with various Kubernetes resources||
+|`LOGDNA_K8S_METADATA_INCLUSION`|Comma separated list of Kubernetes selectors to include log lines associated with various Kubernetes resources||
```
Comment (dkhokhlov) [inline docs/README.md:195]: order of processing of these?

**Review by james-t-bloom — APPROVED** [review]

File: docs/README.md (hunk 3 above)
Comment (dkhokhlov) [reply docs/README.md:195]: line exclusion/inclusion rules have LINE in name - like LOGDNA_LINE_. LOGDNA_K8S_METADATA_EXCLUSION may imply  exclusion of the metadata itself, which is not the case here.

File: common/config/src/argv.rs [hunk 4]
```diff
@@ -147,6 +147,12 @@ pub struct ArgumentOptions {
     #[structopt(long, env = env_vars::LINE_INCLUSION)]
     line_inclusion: Vec<String>,
 
+    #[structopt(long = "k8-metadata-inclusion", env = env_vars::K8S_METADATA_INCLUSION)]
+    k8s_metadata_inclusion: Option<Vec<String>>,
+
+    #[structopt(long = "k8s-metadata-exclusion", env = env_vars::K8S_METADATA_EXCLUSION)]
+    k8s_metadta_exclusion: Option<Vec<String>>,
```
Comment (c-nixon) [inline common/config/src/argv.rs:154]: Minor comment: typo?

File: docs/README.md (hunk 2 above)
Comment (c-nixon) [reply docs/README.md:338]: I still have a slight preference for `name` since that's what it's called in the k8s manifest

**Review by c-nixon — APPROVED** [review]
Couple of minor nits, but this lgtm

File: docs/README.md (hunk 3 above)
Comment (jiangao1) [reply docs/README.md:195]: Also, what is the process we are using MZ_ as a prefix instead of LOGDNA_ ? Just a heads up.

File: docs/README.md [hunk 5]
```diff
@@ -313,6 +313,45 @@ Notes:
 * All regular expressions are case-sensitive by default. If you don't want to differentiate between upper and lower-case letters, use non-capturing groups with a flag: `(?flags:exp)`, for example: `(?i:my_case_insensitive_regex)`
 * LogDNA also provides post-ingestion <a href="https://docs.logdna.com/docs/excluding-log-lines" target="_blank">exclusion rules</a> to control what log data is displayed and stored in LogDNA.
 
+### Configuration for Kubernetes Metadata Filtering
```
Comment (gjanco) [inline docs/README.md:316]: Move this whole section to docs/KUBERNETES.md

Comment (stsantilena) [reply docs/README.md]: Done.

**Pull request merged**